contracts/btc-staking/src/staking.rs

```diff
@@ -322,6 +364,154 @@ pub fn handle_slash_fp(
     slash_finality_provider(deps, env, fp_btc_pk_hex)
 }
 
+pub fn handle_distribute_rewards(
+    mut deps: DepsMut,
+    env: &Env,
+    info: &MessageInfo,
+    rewards: &[RewardInfo],
+) -> Result<Vec<Event>, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // TODO: Check that the sender is the finality contract (AML)
```

Comment:
Wdyt we implement this in this PR? Seems important as otherwise anyone can trigger reward dist anytime

Reply:
Will do.

Reply:
Done in the latest version.